pkg/util/loadbalancer/loadbalancer.go

```diff
@@ -28,3 +31,20 @@ func RemoveFrontendIPConfiguration(lb *armnetwork.LoadBalancer, resourceID strin
 func isFrontendIPConfigReferenced(fipConfig *armnetwork.FrontendIPConfiguration) bool {
 	return fipConfig.Properties.LoadBalancingRules != nil || fipConfig.Properties.InboundNatPools != nil || fipConfig.Properties.InboundNatRules != nil || fipConfig.Properties.OutboundRules != nil
 }
+
+func RemoveHealthProbe(lb *armnetwork.LoadBalancer, resourceID string) error {
+	newProbes := make([]*armnetwork.Probe, 0)
+
+	// Iterate over probes, build a list without the targeted resourceID, if possible
+	for _, probe := range lb.Properties.Probes {
+		if strings.EqualFold(*probe.ID, resourceID) {
+			if probe.Properties.LoadBalancingRules != nil {
+				return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "", fmt.Sprintf("probe %s is used by load balancing rules, remove the referencing load balancing rules before removing the probe", resourceID))
```

Comment:
Small nit: Start sentences with caps letter "Probe", and perhaps use the full name of the resource:"Health Probe %s ...   removing the Health Probe"